### PR TITLE
refactor annotate code once again

### PR DESCRIPTION
**Pull request by vladak**
This change refactors the annotate JSP code once again, the top-level decision making is now checking only:
  - directory
  - revision
  - otherwise (redirect)

### CODE REVIEW

**Comment by vladak** [comment]
After this change the code looks like this:

```java
if (cfg.isDir()) {
  // directory listing
} else if (rev.length() != 0) { // this is true if r parameter is on
  // requesting a revision
  if (cfg.isLatestRevision(rev)) {
    if (cfg.annotate()) {
      // annotate latest revision of the file
    } else {
      // dump the file
    }
  } else {
    // fetch historic revision and dump it (with possible annotation)
  }
} else {
  // redirect to URI with r? parameter
}
```

I believe that in order to dump the historical (annotated) version of the file it is needed to use `AnalyzerGuru.writeXref()` (as opposed to `AnalyzerGuru.writeDumpedXref()`)

File: web/list.jsp [hunk 1]
```diff
@@ -282,8 +285,8 @@ Click <a href="<%= rawPath %>">download <%= basename %></a><%
                                 // SRCROOT is read with UTF-8 as a default.
                                 r = IOUtils.createBOMStrippedReader(in,
                                     StandardCharsets.UTF_8.name());
-                                AnalyzerGuru.writeDumpedXref(
-                                        request.getContextPath(), a, r, out,
+                                AnalyzerGuru.writeXref(
```
Comment (idodeclare) [inline web/list.jsp]: This was not reverted after you found the true problem.

Comment (vladak) [reply web/list.jsp]: Yeah, I was sort of waiting for you to chime in :-) Need to test if the fix still works with `writeDumpedXref()`.

Comment (vladak) [reply web/list.jsp]: I can see that `writeDumpedXref()` is just a wrapper of `writeXref()` so it should work fine.

**Pull request merged**